### PR TITLE
Add wallet uniffi

### PR DESCRIPTION
**Pull request by enzoc4**
This could be used to replace #76 and #88 later.

Potentially, this could be split in two PRs. One that just adds the core api, and another one that makes the changes on the plugin in order to use this. 

I'll leave it up to the reviewers to decide. This looks bigger than it actually is, as a lot of code is either auto-generated, or a semi-automated translation from java to kotlin (by the IDE), or just a port of the existing code, but it may be hard to review as it spans many files. 

Some considerations:

- I don't really know kotlin, so the code is a bit rough yet, and there are also many copies/conversions that can be avoided I think. In general, I didn't focus on performance.
- For the plugin we were passing the pointers directly to javascript, because it was simpler to do. Because the bindings generated by uniffi don't let you get the pointer, I decided to have incremental ids, and put objects in maps instead, which is probably better anyway as its safer.
- Kotlin requires cordova 9, and it's disabled by default. I don't think this is a problem because enabling it is really easy anyway.

### CODE REVIEW

File: bindings/wallet-cordova/tests/src/main.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ PASSWORD[0] = keys.password[0];
 PASSWORD[1] = keys.password[1];
 PASSWORD[2] = keys.password[2];
 PASSWORD[3] = keys.password[3];
-const VOTE_ENCRYPTION_KEY = 'votepk1nc988wtjlrm5k0z43088p0rrvd5yhvc96k7zh99p6w74gupxggtqwym0vm';
+const VOTE_ENCRYPTION_KEY = 'ristretto255_votepk1nc988wtjlrm5k0z43088p0rrvd5yhvc96k7zh99p6w74gupxggtqyx4792';
```
Comment (Zeegomo) [inline bindings/wallet-cordova/tests/src/main.js:17]: oops, this slipt through

Comment (enzoc4) [reply bindings/wallet-cordova/tests/src/main.js:17]: yeah, that happens when things are not ran on ci

**Review by eugene-babichenko — COMMENTED** [review]
Do we really need `Cargo.lock`? It is not a common practice to commit alongside libraries.

Otherwise looking like a great start. I assume that since uniffi can provide bindings for Swift we can finally get away from supporting our small yet annoying Objective-C codebase.

**Comment by enzoc4** [comment]
We can also replace the objective C, yes, it should even be simpler I think.

And... Well, I'm not adding the `Cargo.lock` here, we did it in https://github.com/input-output-hk/chain-wallet-libs/pull/181

The thing is, it's really annoying not to have it because the CI can break at any moment for unrelated reasons.

And also, I generally like to be sure that if we make a release of the plugin then it's using the same `chain-libs` version that it was using when I tested it. 

I think the convention of not adding the lock file for libraries applies more to rust libraries, we are static linking the dependencies anyway, so it's not like the user would have a chance of picking a different version.

**Comment by eugene-babichenko** [comment]
So are we going to remove `wallet-jni` and `wallet-android` bindings as we have the shiny new thing?

**Comment by enzoc4** [comment]
yep, I could do it here too actually. `wallet-android` is unused and out of date, and `wallet-jni` is not that good of a java api anyway.

I feel like we should keep the C api though.

**Comment by eugene-babichenko** [comment]
Yeah, that sounds like a good plan.

**Pull request merged**